### PR TITLE
Add net.blix.BlueMail

### PR DESCRIPTION
**Pull request by Blackburn29**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** __Email sent to Blix a week ago. Have yet to receive a response__
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge). (See Above)
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by JakobDev** [comment]
bot, build net.blix.BlueMail

**Comment by JakobDev** [comment]
Linux is case sensitive. So `net.blix.bluemail` is not the same as `net.blix.BlueMail`. You made that mistake a few times in your manifest.

**Comment by Blackburn29** [comment]
Sorry @JakobDev I meant to push that fix earlier today and hadn't had a chance. Will get it fixed asap

File: net.blix.BlueMail.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  <id>net.blix.BlueMail</id>
+  <metadata_license>MIT</metadata_license>
+  <project_license>Proprietary</project_license>
+  <name>BlueMail</name>
+  <developer_name>Blix Inc.</developer_name>
+  <url type="homepage">https://bluemail.me</url>
+  <summary>BlueMail Email and Calendar App</summary>
+  <description>
+    <p>BlueMail by Blix is a free, beautifully designed, universal email app, capable of managing an unlimited number of mail accounts from various providers, allowing for smart push notifications and group emailing while enabling personalization across multiple email accounts. The BlueMail app connects directly to your mail server and is the perfect replacement for your stock email app.</p>
+    <p>This is the proprietary Blix build of BlueMail, packaged into a Flatpak. This repackaging is not supported by Blix.</p>
```
Comment (barthalion) [inline net.blix.BlueMail.metainfo.xml:12]: This should be changed to follow the disclaimer at the bottom of https://beta.flathub.org/apps/details/com.jetbrains.PyCharm-Professional.

File: net.blix.BlueMail.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+  - --device=all
+  - --filesystem=xdg-config/gtk-3.0
+  - --filesystem=host
+  - --allow=devel
```
Comment (barthalion) [inline net.blix.BlueMail.yaml:19]: Why does it need devel permissions?

File: net.blix.BlueMail.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+base-version: '21.08'
+command: bluemail
+tags: [proprietary]
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+  - --device=all
```
Comment (barthalion) [inline net.blix.BlueMail.yaml:16]: Why does it need access to all devices?

File: net.blix.BlueMail.yaml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  - --filesystem=xdg-config/gtk-3.0
+  - --filesystem=host
+  - --allow=devel
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.freedesktop.Flatpak
```
Comment (barthalion) [inline net.blix.BlueMail.yaml:21]: Same question about org.freedesktop.Flatpak.

File: net.blix.BlueMail.yaml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+command: bluemail
+tags: [proprietary]
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+  - --device=all
+  - --filesystem=xdg-config/gtk-3.0
```
Comment (barthalion) [inline net.blix.BlueMail.yaml:17]: Exposing host's GTK3 config sounds like a can of worms. What is it needed for?

File: net.blix.BlueMail.yaml (hunk 4 above)
Comment (Blackburn29) [reply net.blix.BlueMail.yaml]: Removed

File: net.blix.BlueMail.yaml (hunk 5 above)
Comment (Blackburn29) [reply net.blix.BlueMail.yaml]: I was having some issues with Electron and this was leftover from testing. Sorry about that

File: net.blix.BlueMail.yaml (hunk 3 above)
Comment (Blackburn29) [reply net.blix.BlueMail.yaml]: Removed

File: net.blix.BlueMail.yaml (hunk 2 above)
Comment (Blackburn29) [reply net.blix.BlueMail.yaml]: Left over from testing. Removed.

File: net.blix.BlueMail.yaml [hunk 6]
```diff
@@ -0,0 +1,69 @@
+app-id: net.blix.BlueMail
+default-branch: stable
```
Comment (hfiguiere) [inline net.blix.BlueMail.yaml:2]:
```suggestion
```

**Comment by barthalion** [comment]
bot, build net.blix.BlueMail

File: net.blix.BlueMail.metainfo.xml [hunk 7]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>net.blix.BlueMail</id>
+  <metadata_license>MIT</metadata_license>
+  <project_license>Proprietary</project_license>
```
Comment (JakobDev) [inline net.blix.BlueMail.metainfo.xml:5]:
```suggestion
  <project_license>LicenseRef-proprietar</project_license>
```

**Comment by barthalion** [comment]
bot, build net.blix.BlueMail

File: net.blix.BlueMail.yaml [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+base-version: '21.08'
+command: bluemail
+tags: [proprietary]
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+  - --filesystem=host
```
Comment (hfiguiere) [inline net.blix.BlueMail.yaml:15]: that's a very broad filesystem permission for a proprietary email client.

**Comment by Blackburn29** [comment]
bot, build net.blix.BlueMail

**Comment by Blackburn29** [comment]
bot, build net.blix.BlueMail

**Comment by hfiguiere** [comment]
still can't find the icon

File: net.blix.BlueMail.yaml [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+          - mv opt/BlueMail BlueMail
+          - rm -r BlueMail.deb control.tar.* data.tar.xz debian-binary usr
+      - type: extra-data
+        filename: BlueMail.deb
+        only-arches: [x86_64]
+        url: https://packages.bluemail.me/repos/debian/BlueMail-1.131.12.deb
+        sha256: fee80ffe88ab7ea5d6c3b1b89516bab6718912249dbbfc97ae8a8f30fc8c427e
+        size: 76009334
+        x-checker-data:
+          type: snapcraft
```
[the remark this answers is not in the document]
Comment (Blackburn29) [reply net.blix.BlueMail.yaml:50]: I'll have to update it to do so. Their debian repo is malformed and the update check fails.

**Comment by Blackburn29** [comment]
> still can't find the icon

I honestly don't know why it can't find it. I've messed with it and I'm sure it's something simple but the icon exists when I install my own flatpak locally

File: net.blix.BlueMail.metainfo.xml [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    <p>BlueMail by Blix is a free, beautifully designed, universal email app, capable of managing an unlimited number of mail accounts from various providers, allowing for smart push notifications and group emailing while enabling personalization across multiple email accounts. The BlueMail app connects directly to your mail server and is the perfect replacement for your stock email app.</p>
+    <p>NOTE: This package is not verified by, affiliated with, or supported by Blix Inc.</p>
+  </description>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://www.bluemail.me/img/desktop/Bluemail-Linux-Application.png</image>
+      <caption>BlueMail for Linux</caption>
+    </screenshot>
+  </screenshots>
+  <launchable type="desktop-id">net.blix.bluemail.desktop</launchable>
```
Comment (hfiguiere) [inline net.blix.BlueMail.metainfo.xml:20]: this is case sensitive, therefor it is incorrect, and appstream-compose wil fail causing the build error.

**Comment by Blackburn29** [comment]
bot, build net.blix.BlueMail

**Comment by Blackburn29** [comment]
Looks like the checksum for extra-data has changed. I will get that updated

**Comment by Blackburn29** [comment]
bot, build net.blix.BlueMail

File: net.blix.BlueMail.yaml [hunk 11]
```diff
@@ -0,0 +1,72 @@
+app-id: net.blix.BlueMail
+runtime: org.freedesktop.Sdk
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '21.08'
```
Comment (JakobDev) [inline net.blix.BlueMail.yaml:6]:
```suggestion
runtime-version: '22.08'
sdk: org.freedesktop.Sdk
base: org.electronjs.Electron2.BaseApp
base-version: '22.08'
```

**Comment by Blackburn29** [comment]
bot, build net.blix.BlueMail

**Pull request closed without merging**